dashboard/app/controllers/lessons_controller.rb

```diff
@@ -43,9 +43,11 @@ def update
     end
 
     resources = (lesson_params['resources'] || []).map {|key| Resource.find_by_key(key)}
+    vocabularies = (lesson_params['vocabularies'] || []).map {|key| Vocabulary.find_by_key(key)}
```

Comment:
don't these need to be found by key and course version id?

Reply:
Oh good catch!